File: code/backend-php/src/Core/Port/JwtPort.php
<?php

namespace Games\Paths\Core\Port;

/**
 * Port for JWT token generation — matches Python JwtPort exactly.
 */
interface JwtPort
{
    public function generateAccessToken(string $userUuid, string $username, string $role): string;
    public function generateRefreshToken(string $userUuid): string;
    public function getAccessTokenExpirationMs(): int;
    public function getRefreshTokenExpirationMs(): int;
}

File: code/backend-python/app/core/ports/auth/jwt_port.py
from abc import ABC, abstractmethod

class JwtPort(ABC):
    @abstractmethod
    def generate_access_token(self, user_uuid: str, username: str, role: str) -> str:
        pass

    @abstractmethod
    def generate_refresh_token(self, user_uuid: str) -> str:
        pass

    @abstractmethod
    def get_access_token_expiration_ms(self) -> int:
        pass

    @abstractmethod
    def get_refresh_token_expiration_ms(self) -> int:
        pass
